Source/cmake/BubblewrapSandboxChecks.cmake

```diff
@@ -1,42 +1,41 @@
 if (ENABLE_BUBBLEWRAP_SANDBOX)
-    find_program(BWRAP_EXECUTABLE bwrap)
-    if (NOT BWRAP_EXECUTABLE)
-        message(FATAL_ERROR "bwrap executable is needed for ENABLE_BUBBLEWRAP_SANDBOX")
-    endif ()
-
     find_package(Libseccomp)
     if (NOT Libseccomp_FOUND)
         message(FATAL_ERROR "libseccomp is needed for ENABLE_BUBBLEWRAP_SANDBOX")
     endif ()
 
-    find_program(DBUS_PROXY_EXECUTABLE xdg-dbus-proxy)
-    if (NOT DBUS_PROXY_EXECUTABLE)
-        message(FATAL_ERROR "xdg-dbus-proxy not found and is needed for ENABLE_BUBBLEWRAP_SANDBOX")
+    if (NOT DEFINED BWRAP_EXECUTABLE)
+        if (CMAKE_CROSSCOMPILING)
+            message(FATAL_ERROR "bwrap executable is needed for ENABLE_BUBBLEWRAP_SANDBOX. Unable to autodetect the path when cross-compiling. "
+                                "Please define define the CMake variable BWRAP_EXECUTABLE with the run-time full-path to the 'bwrap' program.")
+        else ()
+            find_program(BWRAP_EXECUTABLE bwrap)
+            if (NOT BWRAP_EXECUTABLE)
+                message(FATAL_ERROR "bwrap executable is needed for ENABLE_BUBBLEWRAP_SANDBOX. "
+                       "Either install it or use the CMake variable BWRAP_EXECUTABLE to define the runtime path.")
+            endif ()
+        endif ()
     endif ()
 
-    if (NOT CMAKE_CROSSCOMPILING)
-        execute_process(
-            COMMAND "${BWRAP_EXECUTABLE}" --version
-            RESULT_VARIABLE BWRAP_RET
-            OUTPUT_VARIABLE BWRAP_OUTPUT
-        )
-        if (BWRAP_RET)
-            message(FATAL_ERROR "Failed to run ${BWRAP_EXECUTABLE}")
-        endif ()
-        string(REGEX MATCH "[0-9]+\\.[0-9]+\\.[0-9]+" BWRAP_VERSION "${BWRAP_OUTPUT}")
-        if (NOT "${BWRAP_VERSION}" VERSION_GREATER_EQUAL "0.3.1")
-            message(FATAL_ERROR "bwrap must be >= 0.3.1 but ${BWRAP_VERSION} found")
```

Comment:
This is removing the version check when not cross-compiling. I would rather leave it and move it to the appropriate location.

Reply:
Yes, that is on purpose. I don't think that version check makes sense anymore. Our oldest distro supported (Ubuntu-20.04) already ships brwap 0.4

Quoting from the changelog

> This patch also removes a version check for bwrap because that is not longer
> needed. The minimum version supported (which is 0.3.1) was released on Sep 2018
> and Ubuntu 20.04 already ships 0.4.0

Reply:
OTOH keeping the check is not only harmless, it may help people who accidentally try to build with older versions. In general we should have all the version checks possible done during the build configuration, if possible.